packages/php-wasm/web-service-worker/src/initialize-service-worker.ts

```diff
@@ -22,28 +33,28 @@ export function initializeServiceWorker(config: ServiceWorkerConfiguration) {
 
 		// Don't handle requests to the service worker script itself.
 		if (url.pathname.startsWith(self.location.pathname)) {
-			return;
+			// return;
 		}
 
 		// Only handle requests from scoped sites.
 		// So – bale out if the request URL is not scoped and the
 		// referrer URL is not scoped.
-		if (!isURLScoped(url)) {
-			let referrerUrl;
-			try {
-				referrerUrl = new URL(event.request.referrer);
-			} catch (e) {
-				return;
-			}
-			if (!isURLScoped(referrerUrl)) {
-				// Let the browser handle uncoped requests as is.
-				return;
-			}
+		if (
+			!isURLScoped(url) &&
+			!(
+				event.request.referrer &&
+				isURLScoped(new URL(event.request.referrer))
+			)
+		) {
+			// Add caching for non-scoped requests
+			event.respondWith(cachedFetch(event.request));
+			return;
 		}
-		const responsePromise = handleRequest(event);
-		if (responsePromise) {
-			event.respondWith(responsePromise);
+		const response = handleRequest(event);
+		if (response) {
+			event.respondWith(response);
```

Comment:
Are these scoped requests for things like WP assets? Does that mean we are not caching them? Or do the de-scoped requests made by `handleRequest()` become new `fetch` events that are handled above.

In my current mental model, fetches by the Service Worker do not trigger additional `fetch` events, but that might just be a poor assumption on my part.

Reply:
I'm not sure if this helpes, but here is how requests work.

All requests in the app go to this listener, no matter if it's a remote request or an internal Playground request.

Remote requests should be cached to add offline support. These are files like the WP zip and remote static assets.
There are also third-party remote requests like Google analytics, that we don't cache for now. 

Internal Playground requests are the same as scoped requests. This is where we reroute the request to go to PHP-wasm instead of going to a fetch. This would be a WordPress page load, an asset that exists inside the Playground filesystem like style.css, or a PHP script execution request.

Reply:
Thank you for explaining a bit. In my mental model yesterday, I'd forgotten that many WordPress files would be available locally.

Note: The following question may be irrelevant if you are planning to proactively cache remote WP assets separately.

For scoped requests for remote static assets that were eliminated from the WordPress build, won't those skip caching because:
- scoped requests are not currently cached
- scoped requests for remote WP assets are fetched by the Service Worker
- fetch() by the Service Worker does not trigger subsequent `fetch` events (in my understanding and testing)
?